### PR TITLE
Add description on Tag form

### PR DESCRIPTION
**Pull request by mzagmajster**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [no]
| New feature/enhancement? (use the a.x branch)      | [yes]
| Deprecations?                          | [no]
| BC breaks? (use the c.x branch)        | [no]
| Automated tests included?              | [no] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

This PR adds summary and description on Tag form so the marketers can describe more precisely what given tag represents.

Note: We are now able to edit tag only the first time we open the form. If we want to rename the tag we have to create a new one, this however is not true for summary and description, we can update those two fields as much as we want (this is the expected behavior, you can read about reasoning behind this decision here: #10927).

I will try to add tests (and probably need some assistance on what scenario do I actually add) if codecov goes down.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Go to new tag form
3. You can now add description to tag.

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10799"></a>

### CODE REVIEW

**Comment by kuzmany** [comment]
What is reason to make two fields for descriptions (description and summary)? From my point of view, description is enough

**Comment by mzagmajster** [comment]
Summary is shown on list view and is short description. Description is
shown on details.

**Comment by kuzmany** [comment]
As you see description is used in most entities (emails, segments, campaign, forms, stages...) and it's integrated without additional coding.. I vote to stay in that line for tags too and avoid differences between them.

**Comment by mzagmajster** [comment]
Not sure what you mean by "integrated without additional coding", I had 
to add migration for description field so that the field can be added 
later. As far as summary field go, I can remove it if that would make it 
possible to accept this PR. I still need summary field to meet client 
requirements, but I guess I can add it via plugin functionality.

Tags originally did not have a description field.

Regards, M.

**Comment by kuzmany** [comment]
I recommend stay in format what we've already use in each section.

Like in segments for example

image

**Comment by mzagmajster** [comment]
Here @kuzmany I have removed summary field from this contribution I hope its OK now, please check it and let me know. Thanks.

**Comment by mzagmajster** [comment]
Not sure why this is failing now. Migration seems fine to me. Can someone provide an insight on why test is failing now?

**Comment by RCheesley** [comment]
Hi @mzagmajster you can see the errors in the run details here: https://github.com/mautic/mautic/runs/5024024604?check_suite_focus=true#step:9:83

**Comment by mzagmajster** [comment]
Hi, @RCheesley thank you, but these errors are not helpful to me. I added a simple migration to add description from the output of the tests its not clear to me how to fix it, it does not even seem to be related to my work.

Hmm... looking and thinking further does it complain the way it does because I fixed the migration I initially added a couple commits before on this branch and then with the latest commit I changed it? Maybe... I will check this out.

**Comment by kuzmany** [comment]
@mzagmajster you need rebase your branch to actual 4.x
If it's more easy to you you can create new PR and cherry pick your commits

PR looks good and it's nice.  After clean up I will tests, hope we get it to 4.2. Thank you

**Review by kuzmany — DISMISSED** [review]
Nice addition.
Works as describe.
We should add description to add/edit/view  functional tests workflow 

- https://github.com/mautic/mautic/blob/dcec76142986c4846cb2218bda6ea81d11a4f554/plugins/MauticTagManagerBundle/Tests/Functional/Controller/TagControllerTest.php#L155
- https://github.com/mautic/mautic/blob/dcec76142986c4846cb2218bda6ea81d11a4f554/plugins/MauticTagManagerBundle/Tests/Functional/Controller/TagControllerTest.php#L104
- https://github.com/mautic/mautic/blob/dcec76142986c4846cb2218bda6ea81d11a4f554/plugins/MauticTagManagerBundle/Tests/Functional/Controller/TagControllerTest.php#L82

**Comment by mzagmajster** [comment]
Hi, @kuzmany 

I tried to rebase this branch with the latest 4.x and check for mariadb is still failing with the same error. I suspect even if I create new PR this will still be the case. What should I do?

**Comment by kuzmany** [comment]
@mzagmajster once If you resolve branch conflicts,  I am able to help you with finish it.

**Comment by jos0405** [comment]
I tested this one, works as expected. I really like this addition.

**Comment by jos0405** [comment]
image

**Review by jos0405 — DISMISSED** [review]
Works as expected.
image

File: app/bundles/LeadBundle/Entity/Tag.php [hunk 1]
```diff
@@ -56,6 +62,8 @@ public static function loadApiMetadata(ApiMetadataDriver $metadata)
                 [
                     'id',
                     'tag',
+                    'summary',
```
Comment (kuzmany) [inline app/bundles/LeadBundle/Entity/Tag.php]: Summary does not exist

File: plugins/MauticTagManagerBundle/Controller/TagController.php [hunk 2]
```diff
@@ -332,9 +332,18 @@ private function createTagModifyResponse(Tag $tag, array $postActionVars, $actio
...
+                    // We are editing existing tag.in the database.
+                    $valid        = true;
+                    $existingTags = $tagModel->getRepository()->getTagsByName([$tag->getTag()]);
+                    foreach ($existingTags as $e) {
+                        if ($e->getId() != $tag->getId()) {
+                            $valid = false;
+                            break;
+                        }
+                    }
+                    //$found = $tagModel->getRepository()->countOccurrences($tag->getTag());
```
Comment (kuzmany) [inline plugins/MauticTagManagerBundle/Controller/TagController.php]: Do not need this line

**Comment by mzagmajster** [comment]
I have cleaned up as the comments suggest.

File: plugins/MauticTagManagerBundle/Controller/TagController.php (hunk 2 above)
Comment (mzagmajster) [reply plugins/MauticTagManagerBundle/Controller/TagController.php]: I have resolved this.

**Review by kuzmany — APPROVED** [review]

**Review by mzagmajster — COMMENTED** [review]
I was able to add description text alongside tag name on tag creation.

**Review by bradycargle — APPROVED** [review]
Everything working for me, nice job!

**Pull request merged**